sycl/include/CL/sycl/detail/program_impl.hpp

```diff
@@ -92,7 +100,20 @@ class program_impl {
     CHECK_OCL_CODE(clGetProgramInfo(ClProgram, CL_PROGRAM_DEVICES,
                                     sizeof(cl_device_id) * NumDevices,
                                     ClDevices.data(), nullptr));
-    Devices = vector_class<device>(ClDevices.begin(), ClDevices.end());
+    vector_class<device> SyclContextDevices = Context.get_devices();
+
+    // Keep only the subset of the devices (associated with context) that
+    // were actually used to create the program.
+    // This is possible when clCreateProgramWithBinary is used.
+    auto iterator = std::remove_if(
```

Comment:
`iterator` is not very clear. Probably you could pick a name from `std::remove_if`documentation.
Perhaps `new_end`?
Hopefully in C++20 with the ranges, this weird erase-remove idiom will disappear...

Reply:
Naming it NewEnd to be consistent with other variables.